fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/exception/MultipleConstraintViolationException.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.api.exception;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Wrapper to hold multiple constraint violation exceptions for later reporting.
+ * @author whikloj
+ */
+public class MultipleConstraintViolationException extends ConstraintViolationException {
+
+    private Set<Exception> exceptionTypes = new HashSet<>();
```

Comment:
Should this be constrained to only `ConstraintViolationException`s?

Reply:
Oh yes, I went back and forth a couple of times. Must have missed this.